codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +401,188 @@ public TextHeaders clear() {
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence value) {
+            CharSequence current = DefaultTextHeaders.super.get(name);
+            if (current == null) {
+                DefaultTextHeaders.super.add(name, commaSeparated(value));
+            } else {
+                DefaultTextHeaders.super.set(name, commaSeparated(current, value));
+            }
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.add(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.add(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object value) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(value));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.setObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.setObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        private CharSequence commaSeparated(Object... values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0 ; i < end; i++) {
+                    singleValue.append(values[i]).append(", ");
```

Comment:
Also note that all values are not of the native type (CharSequence) are serialized/unserialized by the [ValueConverter](https://github.com/netty/netty/blob/master/codec/src/main/java/io/netty/handler/codec/Headers.java#L54) interface.  I think we may need to add an accessor to be able to use this here.  See [DefaultHeaders.java#L518](https://github.com/netty/netty/blob/master/codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java#L518) for a usage example. I think this would apply to both `commaSeparated` methods.  WDYT?